### ISSUE TITLE
[BUG] Playwright opens up a blank page while reusing authentication for multi users using browser context.

### ISSUE THREAD

**Issue by InduKrish** [microsoft/playwright#24134]
Playwright opens up a blank page while reusing authentication for multi users using browser context.
### System info
- Playwright Version: "@playwright/test": "^1.22.1",
- Operating System: mac
- Browser:  chrome
- Other info:

### Source code

I am reusing authentication ( session storage and storage state) to avoid multiple login. and when i run the test, playwright opens a blank page on a new window, then another window with the actual URL. Can you please advise how to get rid of a blank page?

it looks like await context.newPage() and browser.newContext() ---> always create a blank page then opening up the actual page on a different window on a different tab/or the same tab.

https://playwright.dev/docs/auth

1. why adding page fixture, and browser together opens up a blank page? Can you please explain? 

if i remove page, and leaves only browser fixture, it opens a single page and works as expected.

or

```js

   //if you add a page fixture along with browser fixture, it opens up a blank page, without page fixture and only browser, it works ok.
    test('userr test', async ({ browser, page, isMobile }) => {
        const context = await browser.newContext({ storageState: './storage-state2.json' });
        console.log("session storage inside the test:" + process.env.SESSION_STORAGE)

        const sessionStorage = process.env.SESSION_STORAGE;
        await context.addInitScript(storage => {
                const entries = JSON.parse(storage);
                Object.keys(entries).forEach(key => {
                    window.sessionStorage.setItem(key, entries[key]);
                });
        }, sessionStorage);

   
        page = await context.newPage({ignoreHTTPSErrors: true});

        await page.goto(process.env.CREW_PLANNER_BASE_URL);

        await page.getByTestId('menu-option-IFVacationPlannerEntry').click();
        await page.waitForTimeout(2000);
        await page.getByTestId('button-text-card-button-IFVacationAuction').click();
        await expect(page.getByText('Dashboard').first()).toBeVisible()
    });

```

2. and the below code is opening up a blank page in a new tab,

```js
test.use({ storageState: './storage-state2.json'  });
test.only('user test trial', async ({  context, isMobile, screen }) => {
     const sessionStorage = process.env.SESSION_STORAGE;
    await context.addInitScript(storage => {
        const entries = JSON.parse(storage);
        Object.keys(entries).forEach(key => {
            window.sessionStorage.setItem(key, entries[key]);
        });
    }, sessionStorage);

   const page = await context.newPage({ignoreHTTPSErrors: true});
   await page.goto(process.env.CREW_PLANNER_BASE_URL);

    await page.getByTestId('menu-option-IFVacationPlannerEntry').click();
    await page.waitForTimeout(2000);
    await page.getByTestId('button-text-card-button-IFVacationAuction').click();
    await expect(page.getByText('Dashboard').first()).toBeVisible();

    console.log("DONE")
});
```

3.Fixtures: the following code opens two blank pages, and the third page with the actual URL (all three pages are launches on three different windows)

```js
export const test = it.extend({
 crewIFPlannerActor: async ({browser, page}, use) => {
       const context = await browser.newContext({ storageState: './storage-state2.json' });
        const sessionStorage = process.env.SESSION_STORAGE;
        await context.addInitScript(storage => {
            const entries = JSON.parse(storage);
            Object.keys(entries).forEach(key => {
                window.sessionStorage.setItem(key, entries[key]);
            });
        }, sessionStorage);

       page = await context.newPage({ignoreHTTPSErrors: true});

     ///assigning page to a constructor
       const crewIFPlannerActor = UseBidAndAwardActor(page, request, 'IF Planner Actor', `${Env.IF_PLANNER_USERNAME}`, `${Env.IF_PLANNER_PASSWORD}`);
        crewIFPlannerActor.with('page', page);
  
        await use(crewIFPlannerActor);
        await context.close();
    },
});
```

Test: it opens two blank pages like mentioned above
```js
test.only('user test fixture', async ({ crewIFPlannerActor, isMobile }) => {
     await crewIFPlannerActor.goto(Env.CREW_PLANNER_BASE_URL);
     await expect(crewIFPlannerActor.getByText('Dashboard').first()).toBeVisible();
});

```

4.Another fixture example: Here it opens one blank page and one actual page with the URL.

```js
export const test = it.extend({
 crewIFPlannerActor: async ({browser}, use) => {
       const context = await browser.newContext({ storageState: './storage-state2.json' });
        const sessionStorage = process.env.SESSION_STORAGE;
        await context.addInitScript(storage => {
            const entries = JSON.parse(storage);
            Object.keys(entries).forEach(key => {
                window.sessionStorage.setItem(key, entries[key]);
            });
        }, sessionStorage);

      const  page = await context.newPage({ignoreHTTPSErrors: true});

     ///assigning page to a constructor
       const crewIFPlannerActor = UseBidAndAwardActor(page, request, 'IF Planner Actor', `${Env.IF_PLANNER_USERNAME}`, `${Env.IF_PLANNER_PASSWORD}`);
        crewIFPlannerActor.with('page', page);
     
        await use(crewIFPlannerActor);
        await context.close();
    },
});
```
Test for fixture example 4.
```js
test.only('user test fixture', async ({ crewIFPlannerActor, isMobile }) => {
     await crewIFPlannerActor.goto(Env.CREW_PLANNER_BASE_URL);
     await expect(crewIFPlannerActor.getByText('Dashboard').first()).toBeVisible();
});
```

**Comment by mxschmitt** [comment]
1) There you actually have two pages. You have the `page` fixture from Playwright, which depends on the `context` fixture, so it creates a context there and you have a custom context which you create inside your test. This ends up in two windows:

<img width="886" alt="Screenshot 2023-07-11 at 16 30 14" src="https://github.com/microsoft/playwright/assets/17984549/8fa8f91d-9649-4002-8200-a472795a3318">

2) There I recommend not creating your own context. ignoreHTTPSErrors is not a valid option for newPage anyways.This belongs into your playwright.config.ts:

```
  use: {
    ignoreHTTPSErrors: true,
  },
```

```diff
diff --git a/test.js b/test.js
index 082873cf7..4bbea7645 100644
--- a/test.js
+++ b/test.js
@@ -1,5 +1,5 @@
 test.use({ storageState: './storage-state2.json'  });
-test.only('user test trial', async ({  context, isMobile, screen }) => {
+test.only('user test trial', async ({  context, isMobile, screen, page }) => {
      const sessionStorage = process.env.SESSION_STORAGE;
     await context.addInitScript(storage => {
         const entries = JSON.parse(storage);
@@ -8,7 +8,6 @@ test.only('user test trial', async ({  context, isMobile, screen }) => {
         });
     }, sessionStorage);
 
-   const page = await context.newPage({ignoreHTTPSErrors: true});
    await page.goto(process.env.CREW_PLANNER_BASE_URL);
 
     await page.getByTestId('menu-option-IFVacationPlannerEntry').click();
```

3)  Similar thing there, don't use Playwright's page fixture, since you have your own context and page:

```diff
diff --git a/test.js b/test.js
index af860e423..c901d8797 100644
--- a/test.js
+++ b/test.js
@@ -1,5 +1,5 @@
 export const test = it.extend({
-  crewIFPlannerActor: async ({browser, page}, use) => {
+  crewIFPlannerActor: async ({browser}, use) => {
         const context = await browser.newContext({ storageState: './storage-state2.json' });
          const sessionStorage = process.env.SESSION_STORAGE;
          await context.addInitScript(storage => {
@@ -9,7 +9,7 @@ export const test = it.extend({
              });
          }, sessionStorage);
  
-        page = await context.newPage({ignoreHTTPSErrors: true});
+        const page = await context.newPage({ignoreHTTPSErrors: true});
  
       ///assigning page to a constructor
         const crewIFPlannerActor = UseBidAndAwardActor(page, request, 'IF Planner Actor', `${Env.IF_PLANNER_USERNAME}`, `${Env.IF_PLANNER_PASSWORD}`);
```

-> If you use your own context/pages make sure to not use Playwright's page or context fixture by accident, since it will give you an extra window otherwise.

**Comment by InduKrish** [comment]
i am using playwright testing library - screen to perform certain tests, and ensured that page and context are removed from the fixture as advised above, and it still opens up a blank page , Can you please advise how to get rid of the blank page, using the code below? 

```js
crewIFPlannerActor: async ({browser, request, screen, within},use) => {
       const context = await browser.newContext({ storageState: './storage-state2.json' });
  
        const sessionStorage = process.env.SESSION_STORAGE;
        await context.addInitScript(storage => {
            const entries = JSON.parse(storage);
            Object.keys(entries).forEach(key => {
                window.sessionStorage.setItem(key, entries[key]);
            });
        }, sessionStorage);

       const page = await context.newPage({ignoreHTTPSErrors: true});

       //initialize page and screen in the constructor.
        const crewIFPlannerActor = UseBidAndAwardActor(page, request, 'IF Planner Actor', `${Env.IF_PLANNER_USERNAME}`, `${Env.IF_PLANNER_PASSWORD}`,   screen, within);
        crewIFPlannerActor.with('page', page);
        crewIFPlannerActor.with('screen', screen);
        crewIFPlannerActor.with('within', within);

        await use(crewIFPlannerActor);
 
    },
``

```js
 test.only('user test fixture', async ({ crewIFPlannerActor, isMobile }) => {

        console.log("Testing")
       // const page = crewIFPlannerActor.states('page');
        await crewIFPlannerActor.attemptsTo(Navigate.to(Env.CREW_PLANNER_BASE_URL));

        console.log("tree")
        console.log("clicked or not")

        await crewIFPlannerActor.attemptsTo(NavigateToCrewPlanner.menu('Settings & Defaults', isMobile));
    });
```

**Comment by mxschmitt** [comment]
Would it be possible to share a Git repository of your project? With all the code above its hard to follow. Generally speaking I feel like your patterns which you are trying to use makes it harder to follow and debug. We recommend sticking with no specific pattern.

**Comment by InduKrish** [comment]
ok, the repo has user confidential credentials and the URL, Can i share the code without user credential and the URL that you can replace it with your own url and credentials.? Can you please confirm?

and  do you see any problem with the way this fixture is defined? it opens up a blank page. but if i remove screen from the fixture, it doesnt up a blank page.  but i need both screen and page also to be initialized .

crewIFPlannerActor: async ({browser, request, screen, within},use) => {
       const context = await browser.newContext({ storageState: './storage-state2.json' });
  
        const sessionStorage = process.env.SESSION_STORAGE;
        await context.addInitScript(storage => {
            const entries = JSON.parse(storage);
            Object.keys(entries).forEach(key => {
                window.sessionStorage.setItem(key, entries[key]);
            });
        }, sessionStorage);

       const page = await context.newPage({ignoreHTTPSErrors: true});

       //initialize page and screen in the constructor.
        const crewIFPlannerActor = UseBidAndAwardActor(page, request, 'IF Planner Actor', `${Env.IF_PLANNER_USERNAME}`, `${Env.IF_PLANNER_PASSWORD}`,   screen, within);
        crewIFPlannerActor.with('page', page);
        crewIFPlannerActor.with('screen', screen);
        crewIFPlannerActor.with('within', within);

        await use(crewIFPlannerActor);
 
    },
 test.only('user test fixture', async ({ crewIFPlannerActor, isMobile }) => {
        await crewIFPlannerActor.attemptsTo(Navigate.to(Env.CREW_PLANNER_BASE_URL));
        await crewIFPlannerActor.attemptsTo(NavigateToCrewPlanner.menu('Settings & Defaults', isMobile));
    });

**Comment by InduKrish** [comment]
yes will share the screen fixture here shortly, thank you for your assistance again..

**Comment by InduKrish** [comment]
https://www.npmjs.com/package/playwright-testing-library --> playwright testing library provides screen fixture that executes all operations that page fixture does,
eg: screen.goto() etc, 

https://playwright.dev/docs/testing-library --> it has some details on screens as well.

https://testing-library.com/docs/queries/about/#screen -> more details on screen
below is the fixture , i extend playwright testing library and use screen fixture.

```javascript
const base = require("@playwright/test");
import {locatorFixtures as fixtures} from '@playwright-testing-library/test/fixture';  

const it = base.test.extend(fixtures);
const {expect} = it;

import {UseBidAndAwardActor} from '../web/actors/BidAndAwardActors';
import Env from "../web/utils/Env"

export const test = it.extend({
    crewIFPlannerActor: async ({page, browser, request, screen, within}, use) => {
        const crewIFPlannerActor = UseBidAndAwardActor(page, request, 'IF Planner Actor', `${Env.IF_PLANNER_USERNAME}`, `${Env.IF_PLANNER_PASSWORD}`, screen, within);
        crewIFPlannerActor.with('page', page);
        crewIFPlannerActor.with('screen', screen);

        await use(crewIFPlannerActor);
    }
});


export {expect} from '@playwright/test';
```


 const page = await context.newPage({ignoreHTTPSErrors: true});  ---> the way we create context with the new page seems conflicting with screen, but i am not sure if thats truly the case, can you please confirm?

**Comment by mxschmitt** [comment]
We are not familiar with the playwright-testing-library package, since this is a third-party-package. So we can't provide any guidance for that.

Since this issue has now a lot of context and does not contain a clear reproduction yet, I'm going to close it for now. Don't hesitate to re-file with a reproducible which we can run locally. Thanks for your understanding.

**Comment by b-gowri-sankar** [comment]
const directLogin = async () => {
	const browser = await firefox.launch({
		headless: false,
		slowMo: 100,
	});

	try {
		const context = await browser.newContext({
			storageState: "playwright/setup/storage-state.json",
		});
		const page = await context.newPage();

		await page.goto("https://mail.google.com");
		await new Promise((r) => setTimeout(r, 5000));

		await browser.close();
	} catch (error) {
		console.log(error);
		await browser.close();
	}
	// await browser.close();
};

It is opening 3 windows and closing them in the blink of an eye and only 1 mail.google.com windows remains opened(closes 5 seconds after according to the code). could you please tell me how to fix this issue?
